wcfsetup/install/files/lib/system/language/LanguageFactory.class.php

```diff
@@ -45,119 +47,98 @@ protected function init()
      */
     public function getLanguage(int $languageID): ?Language
     {
-        return $this->cache->getCache()->getLanguage($languageID);
+        return $this->cache->getLanguage($languageID);
     }
 
     /**
      * Returns the preferred language of the current user.
-     *
-     * @param int $languageID
-     * @return  Language
      */
-    public function getUserLanguage($languageID = 0)
+    public function getUserLanguage(?int $languageID = null): Language
     {
         if ($languageID) {
-            $language = $this->getLanguage($languageID);
+            $language = $this->cache->getLanguage($languageID);
             if ($language !== null) {
                 return $language;
             }
         }
 
         $languageID = $this->findPreferredLanguage();
 
-        return $this->getLanguage($languageID);
+        return $this->cache->getLanguage($languageID);
```

Comment:
This code can be simplified by inverting the condition in line 58. This way the language only has to be looked up once.

Reply:
Won't this cause that if a languageID is passed to the function which does not exist, it will change its current behavior?

Reply:
Uh, you’re correct. Please disregard my remark.